homeassistant/components/tplink/switch.py

```diff
@@ -40,26 +60,29 @@ async def async_setup_entry(
         coordinator=parent_coordinator,
         feature_type=Feature.Switch,
         entity_class=TPLinkSwitch,
+        new_description_generator=_new_switch_description,
     )
 
     async_add_entities(entities)
 
 
 class TPLinkSwitch(CoordinatedTPLinkEntity, SwitchEntity):
-    """Representation of a feature-based TPLink sensor."""
+    """Representation of a feature-based TPLink switch."""
 
     def __init__(
         self,
         device: Device,
         coordinator: TPLinkDataUpdateCoordinator,
         *,
+        description: TPLinkSwitchEntityDescription,
         feature: Feature,
         parent: Device | None = None,
     ) -> None:
         """Initialize the switch."""
         super().__init__(
             device,
             coordinator,
+            description=description,
```

Comment:
Rather than passing this in, perhaps the base entity should use the `id` to look up if there exists a description in a mapping?

Reply:
Now centralized in `descriptions.py`